frontend/ui/widget/textboxwidget.lua

```diff
@@ -119,9 +119,10 @@ function TextBoxWidget:_splitCharWidthList()
         else
             -- Backtrack the string until the length fit into one line.
             local c = self.char_width_list[idx].char
-            -- We give next char to isSplitable() for a wiser decision
-            local next_c = idx+1 <= size and self.char_width_list[idx+1].char or nil
-            if util.isSplitable(c, next_c) then
+            -- We give next and prev chars to isSplitable() for a wiser decision
+            local next_c = idx+1 <= size and self.char_width_list[idx+1].char or false
+            local prev_c = idx-1 >= 1 and self.char_width_list[idx-1].char or false
```

Comment:
Instead of lookup prev_c from the table every time, can we remember the previous char in a local variable instead?